File: iosApp/iosApp/Models/DeviceModel.swift
//
//  DeviceModel.swift
//  iosApp
//
//  Created by Dmitry Kuklin on 19.06.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation

class DeviceModel: CommonItemModel {
    var type: ItemType
    private(set) var name: String
    private(set) var deviceType: DeviceType
    private(set) var deviceId: String
    private(set) var secretsCount: String
    
    init(name: String, deviceType: DeviceType, deviceId: String, secretsCount: String) {
        self.type = .device
        self.name = name
        self.deviceType = deviceType
        self.deviceId = deviceId
        self.secretsCount = secretsCount
    }
}

enum DeviceType {
    case phone
    case iphone
    case pad
    case noteBook
    
    var image: String {
        switch self {
        case .phone:
            return AppImages.Devices.phoneIco
        case .iphone:
            return AppImages.Devices.iphoneIco
        case .pad:
            return AppImages.Devices.padIco
        case .noteBook:
            return AppImages.Devices.noteBookIco
        }
    }
}

File: androidApp/src/main/java/data/DeviceModel.kt
package data

import android.media.Image
import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.runtime.Immutable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.painter.Painter
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import com.example.metasecret.android.R
import java.util.UUID

@Immutable
data class DeviceModel(
    val name: String,
    val deviceType: DeviceType,
    val deviceId: String,
    val secretsCount: String
) : CommonItemModel {
    override val type: ItemType = ItemType.Device
    override val id: UUID = UUID.randomUUID()
}

enum class DeviceType {
    Phone,
    iPhone,
    Pad,
    NoteBook
}

@Composable
fun getDeviceTypeImage(deviceType: DeviceType): Int {
    return when (deviceType) {
        DeviceType.Phone -> R.drawable.phone_ico
        DeviceType.iPhone -> R.drawable.iphone_ico
        DeviceType.Pad -> R.drawable.tab_ico
        DeviceType.NoteBook -> R.drawable.note_ico
    }
}